### DISCUSSION TITLE
How to integrate submodules from git?

### DISCUSSION THREAD

**Discussion by marco-koenen** [vercel/community#95] [Help] [answered]
Hello everyone,

i am currently developing a project (simple jekyll site) where i need to pull individual components from another repository. I achieve this through submodules. It works as expected when I run `vercel` locally through the CLI, but fails with the Git integration. The integration has all required authorizations for the repository as well. Is there a solution to this that I can pursue? I get the following warning:

`WARN: failed to fetch one or more git submodules:Failed to clone 'data/components' a second time, aborting`

Best wishes,
Marco

**Comment by sejr** [comment]
I'm looking for some help with this too. @marco-koenen were you able to find a solution?

**Reply by marco-koenen** [reply]
I have found an alternative solution [here](https://github.com/vercel/vercel/discussions/4566), which I have not yet tested. If you take that approach, could you please keep me informed?

**Comment by beeinger** [comment, marked as answer]
Vercel as of now does not support private submodules, only public ones, however I have made this workaround:
https://github.com/beeinger/vercel-private-submodule

**Reply by Pieparker** [reply]
Additionally, y'all may wish to follow along with the feature request here: https://github.com/vercel/community/discussions/44

Thanks for reaching out!
